source/bao_hyp/config.rst

```diff
@@ -65,39 +65,404 @@ describe the image running on the VM (see `Guest Image`_);
+- **size** [mandatory] - corresponds to the image size. For builtin images declared using
+  `VM_IMAGE`, this value can be defined using the macro VM_IMAGE_SIZE(img_name);
+- **separately_loaded** [optional] - informs the hypervisor if the VM image is to be loaded
+  separately by a bootloader; By default, separately_loaded is set as false;
+- **inplace** [optional]- use the image inplace and don’t copy the image. By default, inplace is
+  set as false;
+
+To ensure accurate and efficient configuration of VM images, it is strongly recommended to leverage
+the designated macros provided by Bao. These macros, namely ``VM_IMAGE_BUILTIN`` and
+``VM_IMAGE_LOADED``, are specifically designed to simplify the image configuration process and
+enhance compatibility with the hypervisor.
+
+
+1. **VM_IMAGE_BUILTIN** - This macro simplifies image configuration by requiring only the
+   ``img_name`` and the image ``base_addr``. This macro specifies both the base address and image
+   size.
+
+2. **VM_IMAGE_LOADED** - This macro requires additional configurations. It requires the definition
+   of image ``base_addr``, the image ``load_addr``, and the image ``size``.
+
+Using these macros not only streamlines the configuration steps but also ensures adherence to the
+correct syntax and parameters. Attempting to manually configure image details without utilizing
+these macros may result in errors or unintended behavior.
+
+The integration of the appropriate macro tailored to your specific use case, encompassing essential
+parameters such as img_name, base_addr, load_addr, and size. This practice not only fosters
+consistency but also enhances reliability in your VM setup. For instance:
+
+- **IMAGE_BUILTIN**: Simplifies system configuration by leveraging Bao's default knowledge of the
+  image location. No separate configuration or loading of guest images through a bootloader is
+  required, and adjustments to the size of guest images are unnecessary.
+
+- **IMAGE_LOADED**: Highly recommended, especially for MPU systems, where manual allocation of space
+  for the guest image can be challenging if embedded in Bao's binary. Without utilizing LOADED, Bao
+  may need to copy the image, potentially resulting in space wastage
+
+Moreover, if the ``separately_loaded`` parameter is configured as false, the hypervisor interprets
+this setting as the offset of the built-in guest image within its own image, denoted as
+``VM_IMAGE_OFFSET``. During run-time, the hypervisor adjusts this value to be subsequently
+interpreted as a physical address. This adjustment involves adding the address at which the
+hypervisor itself was loaded. However, if the ``separately_loaded`` parameter is configured as
+true, the guest image is not embedded in the hypervisor image; instead, it is loaded independently.
+For more details, refer to the figure below.
+
+.. figure:: img/guest-image.svg
+    :align: center
+    :name: vm-image-fig
```

Comment:
To make the scale to the width of the window or the image seems to small.
```suggestion
.. figure:: img/guest-image.svg
    :align: center
    :width: 100%
    :name: vm-image-fig
```

Reply:
Thanks, fixed!